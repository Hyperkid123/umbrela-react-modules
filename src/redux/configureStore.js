import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
