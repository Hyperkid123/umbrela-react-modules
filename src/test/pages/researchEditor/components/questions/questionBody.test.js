import React from 'react'
import QuestionBody, {QuestionBody as Snapshot} from '../../../../../pages/researchEditorModule/components/questions/questionBody';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Question body component', () => {
  const initialState  = {
    questions: {
      activeQuestion: {
        questionId: 10
      }
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<QuestionBody store={store}/>);
    expect(wrapper).toBeDefined();
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'CloseQuestion',
        }}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should change tab correctly', () => {
    const wrapper = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'CloseQuestion',
        }}
        translate={jest.fn()}
      />
    );
    wrapper.instance().handleTabChange(1);
    expect(wrapper.state().tab).toEqual(1);
  })

  it('should render only preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'OpenQuestion',
        }}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correctly with customHelp', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'CloseQuestion',
          hasCustomHelp: true,
          customHelp: 'foo',
        }}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correctly with scalePoints', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'OrderQuestion',
          scalePoints: 100,
        }}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render correctly with image preview', () => {
    const tree = shallow(
      <Snapshot
        activeQuestion={{
          questionType: 'MediaQuestion',
          url: 'foo.png'
        }}
        translate={jest.fn()}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  })

});
