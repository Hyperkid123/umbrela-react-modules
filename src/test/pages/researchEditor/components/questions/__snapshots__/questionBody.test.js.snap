// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question body component should render correctly empty 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles
    label="undefined: undefined"
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />
  <Connect(QuestionTypeChanger) />
  <WithStyles
    color="default"
    position="static"
  >
    <WithStyles
      centered={true}
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      style={
        Object {
          "marginTop": 5,
        }
      }
      value={0}
    >
      <WithStyles
        color="secondary"
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
      <WithStyles
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
    </WithStyles>
  </WithStyles>
  <styled.div
    column={true}
  >
     
    <Connect(OptionsBody) />
  </styled.div>
</div>
`;

exports[`Question body component should render correctly with customHelp 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles
    label="undefined: undefined"
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />
  <Connect(QuestionTypeChanger) />
  <WithStyles
    color="default"
    position="static"
  >
    <WithStyles
      centered={true}
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      style={
        Object {
          "marginTop": 5,
        }
      }
      value={0}
    >
      <WithStyles
        color="secondary"
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
      <WithStyles
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
    </WithStyles>
  </WithStyles>
  <styled.div
    column={true}
  >
     
    <TextField
      fullWidth={true}
      inputRef={[Function]}
      margin="normal"
      multiline={true}
      name="customHelpInput"
      onBlur={[Function]}
      onChange={[Function]}
      required={false}
      select={false}
      value="foo"
    />
    <Connect(OptionsBody) />
  </styled.div>
</div>
`;

exports[`Question body component should render correctly with image preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles
    label="undefined: undefined"
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />
  <Connect(QuestionTypeChanger) />
  <WithStyles
    color="default"
    position="static"
  >
    <WithStyles
      centered={true}
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      style={
        Object {
          "marginTop": 5,
        }
      }
      value={0}
    >
      <WithStyles
        color="secondary"
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
      <WithStyles
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
    </WithStyles>
  </WithStyles>
  <styled.div
    column={true}
  >
     
    <styled.div
      column={true}
      grow={true}
    >
      <TextField
        fullWidth={true}
        inputRef={[Function]}
        margin="normal"
        name="imagePreviewInput"
        onBlur={[Function]}
        onChange={[Function]}
        required={false}
        select={false}
        value="foo.png"
      />
      <TextFieldComent
        alignRight={true}
        error={true}
      />
    </styled.div>
    <Connect(OptionsBody) />
  </styled.div>
</div>
`;

exports[`Question body component should render correctly with scalePoints 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles
    label="undefined: undefined"
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />
  <Connect(QuestionTypeChanger) />
  <WithStyles
    color="default"
    position="static"
  >
    <WithStyles
      centered={true}
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      style={
        Object {
          "marginTop": 5,
        }
      }
      value={0}
    >
      <WithStyles
        color="secondary"
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
      <WithStyles
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
    </WithStyles>
  </WithStyles>
  <styled.div
    column={true}
  >
     
    <Connect(OptionsBody) />
  </styled.div>
</div>
`;

exports[`Question body component should render only preview 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles
    label="undefined: undefined"
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />
  <Connect(QuestionTypeChanger) />
  <WithStyles
    color="default"
    position="static"
  >
    <WithStyles
      centered={true}
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      style={
        Object {
          "marginTop": 5,
        }
      }
      value={0}
    >
      <WithStyles
        color="secondary"
        fullWidth={true}
        icon={<ShouldUpdate />}
      />
    </WithStyles>
  </WithStyles>
  <styled.div
    column={true}
  >
     
    <Connect(QuestionPreview) />
  </styled.div>
</div>
`;
